v2/cli/init_command.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/brigadecore/brigade/sdk/v2/core"
+	"github.com/brigadecore/brigade/v2/internal/file"
+	"github.com/urfave/cli/v2"
+)
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a new Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to initialize " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage: "Specify the scripting language to use for handling events" +
+				" -- JavaScript (js) or TypeScript (ts)",
+			Value: "ts",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage: "Optionally specify a git repository where the" +
+				" event handling script can be found",
+		},
+	},
+	Action: initialize,
+}
+
+// nolint: gocyclo
+func initialize(c *cli.Context) error {
+
+	// Initialize fields to fill in the template yaml files
+	fields := struct {
+		ProjectID   string
+		Language    string
+		GitCloneURL string
+		Script      string
+	}{
+		c.String(flagID),
+		strings.ToLower(c.String(flagLanguage)),
+		c.String(flagGit),
+		"",
+	}
+
+	err := core.ValidateProjectID(fields.ProjectID)
+	if err != nil {
+		return err
+	}
+
+	// Check if input language is valid
+	fields.Language, err = canonicalizeLanguage(fields.Language)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade directory exists
+	err = file.EnsureDirectory(".brigade")
+	if err != nil {
+		return err
+	}
+
+	// Generate js/ts buffer
+	var scriptBytes []byte
+	scriptBytes, err = execTemplate(scriptTemplate, fields)
+	if err != nil {
+		return err
+	}
+	fields.Script = string(scriptBytes)
+
+	brigadeScriptFileName := fmt.Sprintf("brigade.%s", fields.Language)
+
+	// If git repo is provided, create a separate js/ts file for script
+	var overwrite bool
+	if fields.GitCloneURL != "" {
+		scriptPath := path.Join(".brigade", brigadeScriptFileName)
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
```

Comment:
You don't need the else here. If there was an error, we've returned already. The only way we get to here is if there was no error.